File: projectBlueWater/src/main/java/com/lasthopesoftware/storage/write/exceptions/StorageWriteFileException.java
package com.lasthopesoftware.storage.write.exceptions;

import java.io.File;
import java.io.IOException;

public class StorageWriteFileException extends IOException {

	private final File file;

	public StorageWriteFileException(File file) {
		this(file, null);
	}

	public StorageWriteFileException(File file, Exception innerException) {
		super("There was an error writing the serviceFile " + file + ".", innerException);

		this.file = file;
	}

	public File getFile() { return file; }
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/storage/write/exceptions/StorageWriteFileException.kt
package com.lasthopesoftware.storage.write.exceptions

import java.io.File
import java.io.IOException

open class StorageWriteFileException(
    val file: File? = null,
    innerException: Exception? = null
) : IOException(
	if (file != null) "There was an error writing the file $file."
	else "There was an error writing the file.",
	innerException
)
